test/rational.jl

```diff
@@ -26,4 +26,9 @@
     @test denominator(x / x^2) == x^2
     @test inv(x / x^2) == x
     @test x / x^2 == inv(x)
+    @test isone(((x+1) / (x-1)) / ((x+1) / (x-1)))
+    @test ((x+1)^2 / (x-1)) / ((x+1) / (x-1)) == x+1
+    RType = typeof(x+1/x)
+    @test RType(true) == one(RType)
```

Comment:
Add a comment that this is called from LinearAlgebra in the method `Matrix{T}(s::UniformScaling{Bool}, dims)`. In my opinion they should call `convert(T, s.lambda)` there instead of `T(s.lambda)` now that constructor do not fallback to convert (which used to be the case in Julia v0.6).